### DISCUSSION TITLE
Commission Thread Device to Apple/Google/Amazon BorderRouters

### DISCUSSION THREAD

**Discussion by jseide** [openthread/openthread#10910] [Q&A] [answered]
I'm trying to follow the external commissioning pattern specified here (https://openthread.io/guides/border-router/external-commissioning) to commission a thread device to the network hosted on my apple homepod mini and/or google border router device but don't think I'll be able to without getting the commissioner credential.  

I know IOS/Android have APIs to retrieve credentials, but am confused about their usage. Do I have to use the IOS API to retrieve homepod thread credentials and the Android API to retrieve Google Home thread credentials? So therefore, I need an iphone to act as the commissioner to the homepod and an android to be commissioner to the google home device?  Shouldn't I be able to retrieve the credentials with any commissioner device, including using the ot-commissioner on my laptop to act as the external commissioner?  Is there a facility in the ot-commissioner code to retrieve the commissioner credential?

**Comment by jwhui** [comment]
Today, the only way to extract Thread Network Credentials directly from Apple or Google devices is by using the iOS / Android Thread Network APIs.
- [iOS Thread APIs](https://developer.apple.com/documentation/threadnetwork/)
- [Thread Network SDK for Android](https://developers.home.google.com/thread)

AFAIK, Amazon does make the Thread Network Credentials available via publicly accessible APIs.

That said, other mechanisms for making Thread Network Credentials were specified in both Thread 1.4 and Matter 1.4:
- [Thread 1.4](https://www.threadgroup.org/news-events/blog/ID/875/Thread-14-Paves-The-Path-For-Smart-Devices-To-Work-Together-Regardless-Of-Their-Ecosystem-Or-Manufacturer)
- [Matter 1.4](https://csa-iot.org/newsroom/matter-1-4-enables-more-capable-smart-homes/)

But product implementation will take some time for companies to roll out.

**Reply by jseide** [reply]
So just to confirm- currently IOS API to extract creds from an apple BR and Android API to extract creds from an google BR?

**Reply by jwhui** [reply, marked as answer]
Google Home App on iOS does store Thread Network Credentials using the iOS Thread APIs, so it is possible to retrieve Thread Credentials for Google TBRs on iOS.

Apple Home App does not exist on Android, so iOS is the only way.
